### PR TITLE
AddValidateUpdateItemUsingPath method

### PR DESCRIPTION
**Pull request by koltyakov**
#### Category
- [ ] Bug fix?
- [x] New feature?
- [ ] New sample?
- [ ] Documentation update?

#### Related Issues

mentioned in [490](https://github.com/SharePoint/PnP-JS-Core/issues/490#issuecomment-385398201)

#### What's in this Pull Request?

The PR introduces new endpoint to the library: `AddValidateUpdateItemUsingPath`. Which allows creating items in a specified folder.

#### Guidance

```typescript
const listPath = `${webRelativePath}/Lists/ListUri`;

pnp.sp.web.getList(listPath).addValidateUpdateItemUsingPath([
    { FieldName: 'Title', FieldValue: 'New item in a folder' }
], `${listPath}/FolderPath`).then(console.log);
```

__P.S. Currently the PR is in a draft state. Going to tweak something before the merge.__
Also, a review would be nice.

### CODE REVIEW

File: packages/sp/src/lists.ts [hunk 1]
```diff
@@ -435,6 +435,36 @@ export class List extends SharePointQueryableSecurable {
...
+                formValues,
+                listItemCreateInfo: {
+                    FolderPath: {
+                        DecodedUrl: decodedUrl,
+                        __metadata: { type: "SP.ResourcePath" },
+                    },
+                    __metadata: { type: "SP.ListItemCreationInformationUsingPath" },
+                },
+            }),
+        }).then(res => res.AddValidateUpdateItemUsingPath.results);
```
Comment (patrick-rodgers) [inline packages/sp/src/lists.ts]: Have you tried this in both verbose and non-verbose mode? Usually the method name doesn't come back in minimal.

Comment (koltyakov) [reply packages/sp/src/lists.ts]: You're right. Only checked in a verbose mode, my bad. Just fixed that.

**Comment by ahsanranjha** [comment]
Thank you. I will learn type script from this project, it is good to learn from this perfect code solution.

**Comment by ahsanranjha** [comment]
may i request for minify version?

**Comment by ahsanranjha** [comment]
Yes, Sir. I am learning how to bundle this project to one file like yours this one same 
https://unpkg.com/@pnp/pnpjs@latest/dist/pnpjs.es5.umd.bundle.min.js

**Comment by koltyakov** [comment]
@dudi44,

[Here](https://github.com/koltyakov/sp-sig-20180705-demo/blob/master/src/03-pnp/FieldTypes.md) you can find all data types fingertips.

**Comment by jdela22** [comment]
@koltyakov Hi Andrew, could this work when trying to add multiple items into a folder? I'm successfully creating one item and would like to add up to ~20 at a time. Could we use a batch?

If so, could you provide a simple example I could use as a guide?

Thanks so much

**Comment by koltyakov** [comment]
Hi @jdela22,

Yes, you can use it in batches:

```typescript
import { sp } from '@pnp/sp';

const listPath = '/sites/surveys/Lists/CustomList';

const b = sp.web.createBatch();

sp.web.getList(listPath).inBatch(b).addValidateUpdateItemUsingPath([
    { FieldName: 'Title', FieldValue: 'Item 1' }
], `${listPath}/A_Folder`).then(console.log);

sp.web.getList(listPath).inBatch(b).addValidateUpdateItemUsingPath([
    { FieldName: 'Title', FieldValue: 'Item 2' }
], `${listPath}/A_Folder`).then(console.log);

b.execute().then(() => console.log('Done'));
```

**Comment by phillipharding** [comment]
@koltyakov are [AddValidateUpdateItem()](https://s-kainet.github.io/sp-rest-explorer/#/entity/SP.List/func/AddValidateUpdateItem) and [AddValidateUpdateItemUsingPath()](https://s-kainet.github.io/sp-rest-explorer/#/entity/SP.List/func/AddValidateUpdateItemUsingPath) equivalent to [ValidateUpdateListItem()](https://s-kainet.github.io/sp-rest-explorer/#/entity/SP.ListItem/func/ValidateUpdateListItem) regarding their [SystemUpdate](http://blog.arvosys.com/2018/07/06/list-items-system-update-options-in-sharepoint-online/) characteristics?

**Comment by koltyakov** [comment]
@phillipharding, as far as I know, yes. However, the dates part is always tricky. ;)

**Comment by jdela22** [comment]
@koltyakov, Thank you, that is working as well.

 Intellisense for me is suggesting that the FieldValue can only take strings. I would like to create an item directly in a folder that has number columns and even lookups. Is this the correct function to be using?

Thanks again

**Comment by koltyakov** [comment]
@jdela22,

Yes, FieldValues in this method payload can only be strings, this is correct. Check the [fingerprints](https://github.com/koltyakov/sp-sig-20180705-demo/blob/master/src/03-pnp/FieldTypes.md) for more details.

**Comment by jdela22** [comment]
@koltyakov Amazing, thanks for the link!

**Pull request merged**